beep/tests/test_maccor_to_biologic_mb.py

```diff
@@ -409,6 +415,81 @@ def test_discharge_current_step_conversion(self):
         self.single_step_to_single_seq_test(xml, diff_dict)
         pass
 
+    def test_conversion_with_updated(self):
+        converter = MaccorToBiologicMb()
+
+        with ScratchDir(".") as scratch_dir:
+            reg_params = {
+                'project_name': {0: 'FormDegrade'},
+                'seq_num': {0: 0},
+                'template': {0: 'diagnosticV5.000'},
+                'charge_constant_current_1': {0: 3.0},
+                'charge_percent_limit_1': {0: 30},
+                'charge_constant_current_2': {0: 3.0},
+                'charge_cutoff_voltage': {0: 4.3},
+                'charge_constant_voltage_time': {0: 60},
+                'charge_rest_time': {0: 5},
+                'discharge_constant_current': {0: 2.0},
+                'discharge_cutoff_voltage': {0: 2.7},
+                'discharge_rest_time': {0: 15},
+                'cell_temperature_nominal': {0: 25},
+                'cell_type': {0: 'LiFun240'},
+                'capacity_nominal': {0: 0.240},
+                'diagnostic_type': {0: 'HPPC+RPT'},
+                'diagnostic_parameter_set': {0: 'LiFunForm'},
+                'diagnostic_start_cycle': {0: 100},
+                'diagnostic_interval': {0: 100}
+                          }
+
+            protocol_params_df = pd.DataFrame.from_dict(reg_params)
+            index = 0
+            protocol_params = protocol_params_df.iloc[index]
+            diag_params_df = pd.read_csv(
+                os.path.join(PROCEDURE_TEMPLATE_DIR, "PreDiag_parameters - DP.csv")
+            )
+            diagnostic_params = diag_params_df[
+                diag_params_df["diagnostic_parameter_set"]
+                == protocol_params["diagnostic_parameter_set"]
+                ].squeeze()
+
+            procedure = Procedure.generate_procedure_regcyclev3(index, protocol_params)
+            procedure.generate_procedure_diagcyclev3(
+                protocol_params["capacity_nominal"], diagnostic_params
+            )
+            procedure.set_skip_to_end_diagnostic(4.4, 2.0, step_key="070", new_step_key="095")
```

Comment:
this is the section that confused me

Reply:
Got it. Added some comments to at least highlight what is going on